app/src/main/java/cm/aptoide/pt/download/DownloadFactory.java

```diff
@@ -246,6 +247,31 @@ public Download create(AutoUpdate.AutoUpdateInfo autoUpdateInfo) {
     return download;
   }
 
+  public Download create(DetailedApp app, int downloadAction) {
```

Comment:
Should create a new method we the specific parameters needed to create a Download an reuse everywhere.